src/bootstrap/compile.rs

```diff
@@ -904,6 +904,11 @@ impl Step for Rustc {
             cargo.arg("-p").arg(krate);
         }
 
+        if compiler.stage == 1 {
+            // Relocations are required for BOLT to work.
```

Comment:
I would've expected this if to gate on the usage of BOLT somehow -- do we have a config flag for that? I think just a generic config flag is fine ("enable-bolt-settings") is ok, we don't need it to actually do optimizations. Maybe even "opt-dist" is the right name for it.

Reply:
This is what I have discussed in the PR description and on Zulip. @onur-ozkan has expressed a [desire](https://rust-lang.zulipchat.com/#narrow/stream/326414-t-infra.2Fbootstrap/topic/Adding.20a.20one-off.20linker.20flag/near/392863982) to avoid a BOLT-specific flag, although it seems like the most straightforward solution to me.

Reply:
I don't have problem with doing this with using flags/config.toml. My point was more about its behaviour.

> Add something like --bolt-flags, which would set the flag only for the cases where it's needed (and thus tying its behavior to opt-dist)

Like here, "which would set the flag only for the cases where it's needed" this seems very limited experience to me.

Reply:
Yeah, the interaction of the flag seems very specific. However, I feel like that's also true in general of several other similar flags. For example, the PGO generate/use flags are also only performed in stage 2, and basically I don't think that they are very useful outside of `opt-dist`.

The fact that we pass a specific linker flag is tied to BOLT - that won't really change. The fact that we only pass it to `librustc_driver` is tied to the way the compiler is distributed, and unless that changes, it also don't make sense to do it in any other way. And performing BOLT on earlier stage than 2 also doesn't make sense. So from this point of view, I think that the behavior of the flag is not *that* limited, because it's pretty much the only way we could do it that currently makes sense.

bootstrap and `opt-dist` are intrinsically, for better or worse, so I don't think that this flag will change that or create some unique, "anti-systemic" solution :D